drivers/ethernet/eth_renesas_ra.c

```diff
@@ -0,0 +1,495 @@
+/*
+ * Copyright (c) 2024 Renesas Electronics Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT renesas_ra_ethernet
+
+/* Renesas RA ethernet driver */
+
+#define LOG_MODULE_NAME eth_ra_ethernet
+#define LOG_LEVEL       CONFIG_ETHERNET_LOG_LEVEL
+
+/* Additional configurations to use with hal_renesas */
+#define ETHER_DEFAULT          NULL
+#define ETHER_CHANNEL0         0
+#define ETHER_BUF_SIZE         1536
+#define ETHER_PADDING_OFFSET   1
+#define ETHER_BROADCAST_FILTER 0
+#define ETHER_TOTAL_BUF_NUM    (CONFIG_ETH_RENESAS_TX_BUF_NUM + CONFIG_ETH_RENESAS_RX_BUF_NUM)
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(LOG_MODULE_NAME);
+
+#include <soc.h>
+#include <errno.h>
+#include <zephyr/device.h>
+#include <zephyr/irq.h>
+#include <zephyr/net/net_pkt.h>
+#include <zephyr/net/ethernet.h>
+#include <ethernet/eth_stats.h>
+#include <zephyr/drivers/pinctrl.h>
+#include <zephyr/net/phy.h>
+#include "r_ether.h"
+#include "r_ether_phy.h"
+
+void ether_eint_isr(void);
+void renesas_ra_eth_callback(ether_callback_args_t *p_args);
+static void renesas_ra_eth_buffer_init(const struct device *dev);
+
+extern void ether_init_buffers(ether_instance_ctrl_t *const p_instance_ctrl);
+extern void ether_configure_mac(ether_instance_ctrl_t *const p_instance_ctrl,
+				const uint8_t mac_addr[], const uint8_t mode);
+extern void ether_do_link(ether_instance_ctrl_t *const p_instance_ctrl, const uint8_t mode);
+
+uint8_t g_ether0_mac_address[6] = DT_INST_PROP(0, local_mac_address);
+struct renesas_ra_eth_context {
+	struct net_if *iface;
+	uint8_t mac[6];
+
+	K_KERNEL_STACK_MEMBER(thread_stack, CONFIG_ETH_RA_RX_THREAD_STACK_SIZE);
+	struct k_thread thread;
+	struct k_sem rx_sem;
+	ether_instance_ctrl_t ctrl;
+	/** pinctrl configs */
+	const struct pinctrl_dev_config *pcfg;
+};
+
+struct renesas_ra_eth_config {
+	const ether_cfg_t *p_cfg;
+	const struct device *phy_dev;
+};
+
+#define ETHER_ISR_EE_RECEIVE_EVENT_MASK (0x01070000)
+
+#if ((ETHER_TOTAL_BUF_NUM) >= 2)
```

Comment:
BTW, you could avoid these long list of conditional variable declarations by using `LISTIFY()` macro. It would make the code a bit more readable by avoiding this repetitive code.

Reply:
I changed to use LISTIFY() macro for buffer declare, thank you for your suggestion